experimental/zerotrie/src/serde.rs

```diff
@@ -249,23 +249,25 @@ where
             let lm = LiteMap::<Box<ByteStr>, usize>::deserialize(deserializer)?;
             ZeroTrie::<Vec<u8>>::try_from(&lm)
                 .map_err(D::Error::custom)
-                .map(|trie| trie.map_store(From::from))
+                .map(|trie| trie.cast_store())
         } else {
             // Note: `impl Deserialize for &[u8]` uses visit_borrowed_bytes
             let bytes = <&[u8]>::deserialize(deserializer)?;
             let (tag, trie_bytes) = bytes
                 .split_first()
                 .ok_or(D::Error::custom("expected at least 1 byte for ZeroTrie"))?;
-            let zerotrie =
-                match *tag {
-                    tags::SIMPLE_ASCII => ZeroTrieSimpleAscii::from_store(trie_bytes)
-                        .map_store_into_zerotrie(From::from),
-                    tags::PERFECT_HASH => ZeroTriePerfectHash::from_store(trie_bytes)
-                        .map_store_into_zerotrie(From::from),
-                    tags::EXTENDED_CAPACITY => ZeroTrieExtendedCapacity::from_store(trie_bytes)
-                        .map_store_into_zerotrie(From::from),
-                    _ => return Err(D::Error::custom("invalid ZeroTrie tag")),
-                };
+            let zerotrie = match *tag {
+                tags::SIMPLE_ASCII => ZeroTrieSimpleAscii::from_store(trie_bytes)
```

Comment:
Does this need to use `cast_store`? We have a store, so why not convert and then create the trie, instead of create and then convert.